build-and-deploy-docs/create_mkdocs_config.py

```diff
@@ -0,0 +1,342 @@
+            )
+
+        return urlunparse(link)
+
+    # Iterate through all tokens (including child tokens) to rewrite links and
+    # copy images into the docs/ folder
+    tokens_to_examine = collections.deque(tokens)
+    while tokens_to_examine:
+        token = tokens_to_examine.popleft()
+        if token.children:
+            tokens_to_examine.extend(token.children)
+
+        if token.type == "link_open" and "href" in token.attrs:
+            token.attrs["href"] = sanitize_link(token.attrs["href"])
+
+        elif token.type == "image" and "src" in token.attrs:
+            token.attrs["src"] = sanitize_link(token.attrs["src"])
+
+    # Write out each page to a separate file
+    renderer = mdformat.renderer.MDRenderer()
+    options = {
+        'parser_extension': [
+            mdformat.plugins.PARSER_EXTENSIONS['gfm'],
+            mdformat.plugins.PARSER_EXTENSIONS['tables'],
+        ]
+    }
+
+    table_of_contents = []
+
+    for page in pages:
+        Path(docs_dir, page.get_filename()).write_text(
+            renderer.render(page.tokens, options, {}),
+            encoding="utf-8"
+        )
+
+        table_of_contents.append({page.title: page.get_filename()})
+
+    return table_of_contents
+
+
+def get_mkdocs_config_data(path: Path, repo: str):
+    """ Read the given MKDocs config file or create it from default.
+
+    Args:
+        - `path`: Path to the MKDocs config file. When set to None, default
+          config is used.
+        - `repo`: The github repo name.
+    """
+    pipeline_name = repo.rsplit("/", maxsplit=1)[-1]
+
+    # Make pages link back to the commit that generated these docs (falling
+    # back to `main` if that can't be found)
+    commit_id = os.environ.get("GITHUB_SHA", "main")
+
+    config = {
+        'site_name': pipeline_name,
+        'docs_dir': 'docs/',
+        'repo_url': 'https://github.com/' + repo,
+        'theme': 'readthedocs',
+        'edit_uri_template': f'blob/{commit_id}/README.md',
+        'nav': [],
+    }
+
+    if path:
+        # Update the defaults with the local configuration
+        config.update(yaml.safe_load(path.read_bytes()))
+
+    # Add in the plugins and extensions that we require
+    required_plugins = {"mike"}
+    plugins = config.setdefault("plugins", [])
+    plugins.extend(sorted(required_plugins - set(plugins)))
+
+    required_extensions = {"tables", "admonition"}
+    extensions = config.setdefault("markdown_extensions", [])
+    extensions.extend(sorted(required_extensions - set(extensions)))
+
+    return config
+
+
+def build_mkdocs_config():
+    """ Build the mkdocs config file. """
+    args = parse_args()
+
+    config_data = get_mkdocs_config_data(
+        args.mkdocs_config, args.pipeline_repo)
+
+    # Sanity-check that we're not trying to reach outside the repository
+    if Path(config_data["docs_dir"]).is_absolute():
+        raise ValueError(
+            f"MkDocs doc_dir={config_data['docs_dir']} cannot be absolute!"
```

Comment:
`doc` or `docs`?

Reply:
It is `docs_dir` ([docs doc](https://www.mkdocs.org/user-guide/configuration/#docs_dir)). The code is correct but the error message is wrong.